Code/Framework/AzCore/AzCore/Jobs/Internal/JobTypeEraser.h

```diff
@@ -0,0 +1,229 @@
+            // pointer adjustment.
+            if constexpr (sizeof(Lambda) <= BufferSize && alignof(Lambda) <= alignof(max_align_t))
+            {
+                TypedRelocate(AZStd::forward<Lambda>(lambda), m_buffer);
+                m_lambda = m_buffer;
+            }
+            else
+            {
+                // Lambda has spilled to the heap (or requires extended alignment)
+                m_lambda = reinterpret_cast<char*>(azmalloc(sizeof(Lambda), alignof(Lambda)));
+                TypedRelocate(AZStd::forward<Lambda>(lambda), m_lambda);
+            }
+        }
+
+        TypeErasedJob(TypeErasedJob&& other) noexcept;
+
+        TypeErasedJob& operator=(TypeErasedJob&& other) noexcept;
+
+        ~TypeErasedJob();
+
+        void Link(TypeErasedJob& other);
+
+        // Indicates if this job is a root of the graph (with no dependencies)
+        bool IsRoot();
+
+        void AttachToJobGraph(CompiledJobGraph& graph) noexcept
+        {
+            m_graph = &graph;
+        }
+
+        void Invoke()
+        {
+            m_invoker(m_lambda);
+        }
+
+        uint8_t GetPriorityNumber() const
+        {
+            return static_cast<uint8_t>(m_descriptor.priority);
+        }
+
+    private:
+        friend class CompiledJobGraph;
+        friend class JobWorker;
+
+        // This relocation avoids branches needed if the lambda type is unknown
+        template <typename Lambda>
+        void TypedRelocate(Lambda&& lambda, char* destination)
+        {
+            if constexpr (AZStd::is_trivially_move_constructible_v<Lambda>)
+            {
+                memcpy(destination, reinterpret_cast<char*>(&lambda), sizeof(Lambda));
+            }
+            else if constexpr (AZStd::is_move_constructible_v<Lambda>)
+            {
+                new (destination) Lambda{ AZStd::move(lambda) };
+            }
+            else if constexpr (AZStd::is_copy_constructible_v<Lambda>)
+            {
+                new (destination) Lambda{ lambda };
+            }
+            else
+            {
+                static_assert(
+                    false,
+                    "Job lambdas must be either move or copy constructible. Please verify that all captured data is move or copy "
+                    "constructible.");
+            }
+        }
+
+        // Small buffer optimization for lambdas. We cover our bases here by enforcing alignment on the
+        // class to equal the alignment of the largest scalar type available on the system (generally
+        // 16 bytes).
+        char m_buffer[BufferSize];
+
+        // This value is an offset in a buffer that stores dependency tracking information.
+        uint32_t m_successorOffset = 0;
+        uint32_t m_inboundLinkCount = 0;
+        uint32_t m_outboundLinkCount = 0;
+
+        // May point to the inlined payload buffer, or heap
+        char* m_lambda = nullptr;
+
+        CompiledJobGraph* m_graph = nullptr;
```

Comment:
NOTE: the semantics around how this is destroyed are broken in this early feedback PR (so known bug, just ignore please, have a plan to fix) :)